### PR TITLE
fix: changed the button lable from Upload to Browse

### LINKED ISSUE #9274: Bug - File upload multiple - allow ability to change button text
The 'Upload' button text in the ['File upload - multiple' PF component](https://www.patternfly.org/v4/components/file-upload---multiple#examples) is [hardcoded](https://github.com/patternfly/patternfly-react/blob/004483dd6c2729052170b8fe72faa1658fa2bd2a/packages/react-core/src/components/MultipleFileUpload/MultipleFileUploadButton.tsx) and cannot be changed. 

I expect to be able to change the button text. E.g., label it as 'Browse' instead of 'Upload'. 

Currently, this is not a blocker. 

I am trying to use this component for a feature in Apicurio Registry. We are expecting to implement this feature in Q4.

### PR DESCRIPTION
**Pull request by tarunsamanta2k20**
**What**: Closes #9274 

**Issue Description:**

The 'Upload' button text in the ['File upload - multiple' PF component](https://www.patternfly.org/v4/components/file-upload---multiple#examples) is [hardcoded](https://github.com/patternfly/patternfly-react/blob/004483dd6c2729052170b8fe72faa1658fa2bd2a/packages/react-core/src/components/MultipleFileUpload/MultipleFileUploadButton.tsx) and cannot be changed.

I expect to be able to change the button text. E.g., label it as 'Browse' instead of 'Upload'.

Currently, this is not a blocker.

I am trying to use this component for a feature in Apicurio Registry. We are expecting to implement this feature in Q4.

**Additional issues**:

### CODE REVIEW

File: packages/react-core/src/components/MultipleFileUpload/MultipleFileUploadButton.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ export const MultipleFileUploadButton: React.FunctionComponent<MultipleFileUploa
   return (
     <div className={css(styles.multipleFileUploadUpload, className)} {...props}>
       <Button variant="secondary" aria-label={ariaLabel} onClick={open}>
-        Upload
+        Browse
```
Comment (tlabaj) [inline packages/react-core/src/components/MultipleFileUpload/MultipleFileUploadButton.tsx:24]: We need to add a prop for this so the text can be changed by the consumer and not hard coded.

**Review by tlabaj — CHANGES_REQUESTED** [review]
see inline comment

File: packages/react-core/src/components/MultipleFileUpload/MultipleFileUploadButton.tsx (hunk 1 above)
Comment (tarunsamanta2k20) [reply packages/react-core/src/components/MultipleFileUpload/MultipleFileUploadButton.tsx]: Should I set its default value to 'Browse' if no label has been provided?

File: packages/react-core/src/components/MultipleFileUpload/MultipleFileUploadButton.tsx [hunk 2]
```diff
@@ -9,19 +9,21 @@ export interface MultipleFileUploadButtonProps extends React.HTMLProps<HTMLDivEl
   className?: string;
   /** Aria-label for the button */
   'aria-label'?: string;
+  'label'?: string;
 }
 
 export const MultipleFileUploadButton: React.FunctionComponent<MultipleFileUploadButtonProps> = ({
   className,
   'aria-label': ariaLabel,
+  label,
```
Comment (thatblindgeye) [inline packages/react-core/src/components/MultipleFileUpload/MultipleFileUploadButton.tsx:18]:
```suggestion
  label = "Upload",
```

Right now the button is rendered without any text label

File: packages/react-core/src/components/MultipleFileUpload/MultipleFileUploadButton.tsx [hunk 3]
```diff
@@ -9,19 +9,21 @@ export interface MultipleFileUploadButtonProps extends React.HTMLProps<HTMLDivEl
   className?: string;
   /** Aria-label for the button */
   'aria-label'?: string;
+  'label'?: string;
```
Comment (thatblindgeye) [inline packages/react-core/src/components/MultipleFileUpload/MultipleFileUploadButton.tsx:12]: We'll need a description of the prop above the name. Maybe "Visible text label for the button"

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
It'd also be worth including a check to ensure either an aria-label or label is passed in. Our [NavGroup](https://github.com/patternfly/patternfly-react/blob/6b44876bfb15c8335fc6e263d63f41f55ae8636b/packages/react-core/src/components/Nav/NavGroup.tsx#L24) component has a console warn for something similar

File: packages/react-core/src/components/MultipleFileUpload/MultipleFileUploadButton.tsx [hunk 4]
```diff
@@ -9,15 +9,20 @@ export interface MultipleFileUploadButtonProps extends React.HTMLProps<HTMLDivEl
   className?: string;
   /** Aria-label for the button */
   'aria-label'?: string;
+  /** Visible text label for the button */
   'label'?: string;
```
Comment (tlabaj) [inline packages/react-core/src/components/MultipleFileUpload/MultipleFileUploadButton.tsx:13]: Can we call this `text` or `browseButtonText` for consistency.  We call it `browseButtonText` in the `FileUpload` component.

**Review by tlabaj — CHANGES_REQUESTED** [review]

File: packages/react-core/src/components/MultipleFileUpload/MultipleFileUploadButton.tsx (hunk 4 above)
Comment (tarunsamanta2k20) [reply packages/react-core/src/components/MultipleFileUpload/MultipleFileUploadButton.tsx:13]: are you asking me to change the **'label' to `browseButtonText`?**

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
In addition to the above, since MultipleFileUploadButton is only rendered internally within MultipleFileUploadMain and not exported we'll need to update the MultipleFileUploadMain component to add the new prop and pass it in to where `<MultipleFileUploadButton />` is rendered.

Might be better to rename the new `label` prop to `browseButtonText` since that'd be more descriptive within MultipleFileUploadMain.

File: packages/react-core/src/components/MultipleFileUpload/__tests__/MultipleFileUploadButton.test.tsx [hunk 5]
```diff
@@ -17,4 +17,9 @@ describe('MultipleFileUploadButton', () => {
     const { asFragment } = render(<MultipleFileUploadButton aria-label="test">Foo</MultipleFileUploadButton>);
     expect(asFragment()).toMatchSnapshot();
   });
+
+  test('renders with label applied to the button', () => {
+    const { asFragment } = render(<MultipleFileUploadButton label="test">Foo</MultipleFileUploadButton>);
```
Comment (thatblindgeye) [inline packages/react-core/src/components/MultipleFileUpload/__tests__/MultipleFileUploadButton.test.tsx:22]:
```suggestion
    const { asFragment } = render(<MultipleFileUploadButton browseButtonText ="test">Foo</MultipleFileUploadButton>);
```

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
This is looking good! The test file needed the prop updated. As mentioned above, could you also add the `browseButtonText` prop to the MultipleFileUploadMain component (we could give it the same default value of "Upload"). Then you'll want to pass MultipleFileUploadMain's `browseButtonText` prop into line 40ish where we're rendering `<MultipleFileUploadButton />`.

**Comment by tarunsamanta2k20** [comment]
if everything has been done for this PR, then please merge. if any changes are required then please let me know.

**Review by thatblindgeye — APPROVED** [review]
Thanks for working on this!

**Review by jenny-s51 — APPROVED** [review]

**Review by tlabaj — APPROVED** [review]

**Pull request merged**